src/store_client.cc

```diff
@@ -622,28 +660,34 @@ store_client::readHeader(char const *buf, ssize_t len)
     if (len < 0)
         return fail();
 
-    if (!unpackHeader(buf, len)) {
-        fail();
-        return;
-    }
+    MemObject *const mem = entry->mem_obj;
+    const auto initialRead = (mem->swap_hdr_sz == 0);
+
+    if (initialRead && !unpackHeader(buf, len))
+        return fail();
```

Comment:
I am not sure, but I suspect these changes may go away if my suspicions regarding "swapout header" and "HTTP header" confusion were correct.

Reply:
I tried to clarify the "swapout header" confusion elsewhere.

Reply:
I will revisit this in a new change request.